### PR TITLE
Add errorsFirst option for sorting

### PR DESCRIPTION
**Pull request by Alea81**
We need the date ordering, but the failed builds / test runs should have prio in the ordering over the succeeded once. We need this so that we can react and handle build/test errors as fast as possible.

### CODE REVIEW

File: app/monitor.js [hunk 1]
```diff
@@ -15,7 +15,25 @@ var async = require('async'),
...
+                if (newBuilds[i].hasErrors)
+                    errorChunk.push(newBuilds[i]);
+                else
+                    normalChunk.push(newBuilds[i]);
+            }
+
+            sortBuilds(errorChunk, sortOrder);
+            sortBuilds(normalChunk, sortOrder);
+
+            return errorChunk.concat(normalChunk);
```
Comment (marcells) [inline app/monitor.js]: In that way, the `sortOrder` configuration will be ignored. This behavior is very complicated for a user to understand. I think the new sorting code could be moved after the existing sorting-option. So, `newBuilds` will correctly be sorted by date or project, and then the erroneous builds will be moved to the beginning (if configured).

**Comment by marcells** [comment]
Hi, thanks for your change, but we have little issue here, what could be easily be fixed.

File: app/monitor.js (hunk 1 above)
Comment (Alea81) [reply app/monitor.js]: I've moved it to the end of the function.
Actually, sortOrder was not ignored, the old functionality was not returning the value just sorting it, but you're right, this is confusing to use it like that.

**Comment by Alea81** [comment]
Rebased to your master, should be fine. :)

**Comment by marcells** [comment]
Thank you!

**Pull request merged**